### DISCUSSION TITLE
Post data to Spring with retrofit

### DISCUSSION THREAD

**Discussion by Creezen** [square/retrofit#3971] [Q&A]
I have  a Android project for client and a Spring project for server.

In Android, I want to post the String param "type" and a POJO "User" with  to my Spring project, what is the best way?

The param "type" is used to decide my server function, eg: for query or compare.

Thus, It's used for only once,  it's impossible to put it into the POJO "User"

**Comment by JakeWharton** [comment]
The easiest way to get an answer to a question like "How do I make a request like XYZ" is to provide an example of the fully-formed HTTP request that you expect. From your question it's not clear where the values "type" and "user" go. Are they query parameters, form parameters, multipart parts, serialized in the body, or some combination? Given an example HTTP request it's pretty easy for people here or on StackOverflow to work backwards and help you form the associated Retrofit method and annotations.
